src/Microsoft.ML.Data/DataView/TypedCursor.cs

```diff
@@ -217,6 +217,118 @@ public RowCursor<TRow>[] GetCursorSet(Func<int, bool> additionalColumnsPredicate
                  .ToArray();
         }
 
+        public static void GetVectorAndItemType(string name, Type rawType, IEnumerable<Attribute> attributes, out bool isVector, out Type itemType, IDataView data, string singleName)
+        {
```

Comment:
The bulk of this new code seems to be copied from elsewhere. Can you please try to refactor this better so that we can reuse the code from where it already exists?

Reply:
Also, all this validation need not happen here. We only need to validate the user type.

---
In reply to: [428309799](https://github.com/dotnet/machinelearning/pull/5114#discussion_r428309799) [](ancestors = 428309799)

Reply:
fixed